.travis.yml

```diff
@@ -1,15 +1,46 @@
-language: node_js
-sudo: required
-dist: trusty
-before_script:
-  - npm run compile
-before_install:
-  - docker pull selenium/standalone-chrome:3.11.0-californium
-  - docker run -d -p 4444:4444 selenium/standalone-chrome:3.11.0-californium
-script:
-  - npm test
-  - npm run danger
+language: generic
+sudo: required 
+services: 
+  - docker
 branches:
   only:
     - master
 cache: false
+script:
+  - 'cp ./config/docker/ci-docker-compose.yml ./docker-compose.yml'
+  - 'docker-compose pull'
+  - "docker-compose run -e DANGER_GITHUB_API_TOKEN=$DANGER_GITHUB_API_TOKEN -e TRAVIS=$TRAVIS -e CI=$CI -e HAS_JOSH_K_SEAL_OF_APPROVAL=$HAS_JOSH_K_SEAL_OF_APPROVAL -e TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST -e TRAVIS_REPO_SLUG=$TRAVIS_REPO_SLUG test npm run $TEST_SUITE"
+env:
+  global:
+    - DOCKER_IMAGE='ryanthemanuel/terra-toolkit'
+  matrix:
+    - TEST_SUITE=lint
+    - TEST_SUITE=jest
+    - TEST_SUITE=nightwatch
+    - TEST_SUITE=wdio
+    - TEST_SUITE=danger
+jobs:
+  include:
+    - stage: release terra-node
+      script:
+        - echo "$DOCKER_PASSWORD" | docker login -u "ryanthemanuel" --password-stdin
+        - 'docker build -t ryanthemanuel/terra-node -f ./docker/terra-node/Dockerfile ./docker/terra-node'
+        - 'docker push ryanthemanuel/terra-node'
+      if: type != pull_request
+    - stage: release snapshot of terra-node
+      script:
+        - echo "$DOCKER_PASSWORD" | docker login -u "ryanthemanuel" --password-stdin
+        - "docker build -t ryanthemanuel/terra-node:$TRAVIS_PULL_REQUEST_BRANCH -f ./docker/terra-node/Dockerfile ./docker/terra-node"
+        - "docker push ryanthemanuel/terra-node:$TRAVIS_PULL_REQUEST_BRANCH"
```

Comment:
Do we want the snapshot versions to be associated to the branch?  Or just a generic snapshot?  Also, note to self, I'll need this to be cerner rather than ryanthemanuel